pir/cpp/client.cpp

```diff
@@ -162,23 +163,37 @@ StatusOr<std::vector<int64_t>> PIRClient::ProcessResponse(
   return result;
 }
 
-StatusOr<std::vector<string>> PIRClient::ProcessResponseString(
+StatusOr<std::vector<string>> PIRClient::ProcessResponse(
+    const std::vector<std::size_t>& indexes,
     const Response& response_proto) const {
+  if (indexes.size() != response_proto.reply_size()) {
+    return InvalidArgumentError(
+        "Number of indexes must match number of replies");
+  }
+
+  ASSIGN_OR_RETURN(auto pirdb, PIRDatabase::Create(context_->Params()));
+  StringEncoder encoder(context_->SEALContext());
```

Comment:
Can StringEncoder be a member of the class or of the context directly? Seems binded to the context anyway.

Reply:
It could, but since it's only ever used in ProcessResponse, seems better to me to keep it local. Makes the usage of it more obvious.